api/management.openapi.yaml

```diff
@@ -0,0 +1,238 @@
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+tags:
+  - name: applications
+    description: Cartesi applications
+paths:
+  /applications:
+    get:
+      tags:
+        - applications
+      summary: Fetches list of applications managed by the node
+      description: List of managed applications
+      operationId: getApplications
+      parameters:
+        - name: status
+          in: query
+          description: Filter by application status
+          schema:
+            $ref: "#/components/schemas/ApplicationStatus"
+        - name: offset
+          in: query
+          description: Items to skip for pagination
+          required: false
+          schema:
+            type: integer
+            minimum: 0
+            default: 0
+        - name: limit
+          in: query
+          description: Maximum number of items per page
+          required: false
+          schema:
+            type: integer
+            minimum: 1
+            default: 100
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/PaginatedResult"
+                  - type: object
+                    properties:
+                      data:
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/Application"
+                    required:
+                      - data
+  /applications/{address}:
+    get:
+      tags:
+        - applications
+      summary: Find application by address
+      description: Returns a single application by its address
+      operationId: getApplicationByAddress
+      parameters:
+        - name: address
+          in: path
+          description: Address of application to return
+          required: true
+          schema:
+            $ref: "#/components/schemas/Address"
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Application"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "404":
+          $ref: "#/components/responses/NotFound"
+    put:
+      tags:
+        - applications
+      summary: Adds an application
+      description: Adds an application by its address
+      operationId: addApplication
+      parameters:
+        - name: address
+          in: path
+          description: Application address
+          required: true
+          schema:
+            $ref: "#/components/schemas/Address"
+      requestBody:
+        $ref: "#/components/requestBodies/Application"
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Application"
+        "304":
```

Comment:
I think I'll remove this 304. It's not common to return 304 on a PUT request. It's usually used on a GET with `If-Modified-Since`.
I think the correct way is to just return 200 if the resource is exactly the same.